src/api/request-validations.ts

```diff
@@ -35,6 +35,19 @@ export const validateBodyEquals = (request: RequestExt, expectedBody: unknown |
   }
 };
 
+export const validateCoercedBodyEquals = (request: RequestExt, expectedBody: unknown | undefined): void => {
+  if (expectedBody == null) {
+    if (!isBodyEmpty(request.rawBody)) {
+      throw new ValidationError(BODY_NOT_EQUAL_ERROR_MESSAGE, expectedBody, request.rawBody);
+    }
+    return;
+  }
+
+  if (!deepEqual(coerceDate(request.body), expectedBody, { strict: true })) {
```

Comment:
this is just a string right?
```suggestion
  if (coerceDate(request.body) !== expectedBody) {
```

Reply:
No, it is an object

Reply:
oh I see, does it deeply

Reply:
An example is:
```js
const expectedBody = {
  id: 1,
  name: "Lisa",
  friendly: true,
  birthdate: "2017-12-13T02:29:51Z",
  complexProperty: {
    color: "Red",
  },
};
```
It have two layers.
Why not we compare deeply in case in the future there might be some cases could reuse this?

Reply:
yeah, it is good as it is, resolved the comment